validator/lib/validateExtensions.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+const Cesium = require('cesium');
+
+const defaultValue = Cesium.defaultValue;
+const defined = Cesium.defined;
+
+module.exports = validateExtensions;
+
+const EMPTY_ARRAY = [];
+
+const requiredExtensions = ['3DTILES_content_gltf'];
+
+/**
+ * Check if a tileset's extensions are valid.
+ *
+ * @param {Object} options An object with the following properties:
+ * @param {Object} options.tileset The tileset JSON.
+ * @returns {String} An error message if validation fails, otherwise undefined.
+ */
+function validateExtensions(options) {
+    const tileset = options.tileset;
+    const extensions = tileset.extensions;
+    const extensionsUsed = defaultValue(tileset.extensionsUsed, EMPTY_ARRAY);
+    const extensionsRequired = defaultValue(tileset.extensionsRequired, EMPTY_ARRAY);
+
+    if (defined(extensions)) {
+        for (const extensionName in extensions) {
+            if (extensions.hasOwnProperty(extensionName)) {
+                const extension = extensions[extensionName];
+                if (!extensionsUsed.includes(extensionName)) {
+                    return `${extensionName} must be included in extensionsUsed`;
+                }
+                if (requiredExtensions.includes(extensionName) && !extensionsRequired.includes(extensionName)) {
+                    return `${extensionName} must be included in extensionsRequired`;
+                }
+
+                let message;
+                if (extensionName === '3DTILES_content_gltf') {
+                    message = validate3DTilesContentGltf(extension);
+                }
+                if (defined(message)) {
+                    return `Error in ${extensionName}: ${message}`;
+                }
+            }
+        }
+    }
+}
+
+function validate3DTilesContentGltf(extension) {
+    const gltfExtensionsUsed = defaultValue(extension.extensionsUsed, EMPTY_ARRAY);
+    const gltfExtensionsRequired = defaultValue(extension.extensionsRequired, EMPTY_ARRAY);
+
+    if (!Array.isArray(gltfExtensionsUsed)) {
```

Comment:
I think we can simplify this further using:

```javascript
if (!Array.isArray(gltfExtensionsUsed) || gltfExtensionsUsed.filter(i => typeof i === 'string').length) {
  return 'extensionsUsed must be an array of strings';
}
```

Reply:
I refactored the code a bit so everything calls `validateExtensionsArray`. I kept the code similar to before for readability but now it only happens in one place.